### PR TITLE
chore: fix zone dotted highlight is not showing up on new anvil apps 

### LINKED ISSUE #35231: [Bug]: Zone dotted highlight is not showing up on new anvil apps
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Zone dotted highlight is not showing up on new anvil apps

image

### Steps To Reproduce

1. Add 2 zones to a section
2. Use space distribution handles to move it around
3. Notice the dotted highlight is missing

### Public Sample App

_No response_

### Environment

Production

### Severity

High (Blocker to building or releasing)

### Issue video log

_No response_

### Version

cloud

### PR DESCRIPTION
**Pull request by jsartisan**
Fixes #35231

/ok-to-test tags="@tag.Anvil"

Before:
CleanShot 2024-08-20 at 16 52 13

After:
CleanShot 2024-08-20 at 16 52 31

## Summary by CodeRabbit

- **New Features**
	- Improved widget border style handling by simplifying parameters and logic.
	- Enhanced visual consistency by removing the `elevatedBackground` consideration from various components.
	- Refined test logic to focus on section properties during space distribution.
- **Bug Fixes**
	- Addressed potential inconsistencies in widget appearance related to background elevation settings.

> [!CAUTION]
> 🔴 🔴 🔴 Some tests have failed.
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/10503394684>
> Commit: 574aab588e35fc90e59720010e2b8a249bf824ed
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=10503394684&attempt=1&selectiontype=test&testsstatus=failed&specsstatus=fail" target="_blank">Cypress dashboard</a>.
> Tags: @tag.Anvil
> Spec: 
> The following are new failures, please fix them before merging the PR: <ol>
> <li>cypress/e2e/Regression/ClientSide/Anvil/Widgets/AnvilButtonWidgetSnapshot_spec.ts
> <li>cypress/e2e/Regression/ClientSide/Anvil/Widgets/AnvilCheckboxGroupWidgetSnapshot_spec.ts
> <li>cypress/e2e/Regression/ClientSide/Anvil/Widgets/AnvilCheckboxWidgetSnapshot_spec.ts
> <li>cypress/e2e/Regression/ClientSide/Anvil/Widgets/AnvilCurrencyInputWidgetSnapshot_spec.ts
> <li>cypress/e2e/Regression/ClientSide/Anvil/Widgets/AnvilHeadingWidgetSnapshot_spec.ts
> <li>cypress/e2e/Regression/ClientSide/Anvil/Widgets/AnvilIconButtonWidgetSnapshot_spec.ts
> <li>cypress/e2e/Regression/ClientSide/Anvil/Widgets/AnvilInlineButtonWidgetSnapshot_spec.ts
> <li>cypress/e2e/Regression/ClientSide/Anvil/Widgets/AnvilInputWidgetSnapshot_spec.ts
> <li>cypress/e2e/Regression/ClientSide/Anvil/Widgets/AnvilParagraphWidgetSnapshot_spec.ts
> <li>cypress/e2e/Regression/ClientSide/Anvil/Widgets/AnvilPhoneInputWidgetSnapshot_spec.ts</ol>
> <a href="https://internal.appsmith.com/app/cypress-dashboard/identified-flaky-tests-65890b3c81d7400d08fa9ee3?branch=master" target="_blank">List of identified flaky tests</a>.
> <hr>Thu, 22 Aug 2024 07:30:08 UTC

### CODE REVIEW

File: app/client/src/layoutSystems/anvil/common/hooks/useWidgetBorderStyles.ts [hunk 1]
```diff
@@ -44,14 +40,13 @@ export function useWidgetBorderStyles(
   if (isPreviewMode) {
     return {};
   }
-  const isZoneDistributingSpace =
```
Comment (jsartisan) [inline app/client/src/layoutSystems/anvil/common/hooks/useWidgetBorderStyles.ts:47]: widgetId is of a section when distributing space. Not sure why we were checking zone.
@riodeuno

File: app/client/src/layoutSystems/anvil/common/hooks/useWidgetBorderStyles.ts [hunk 2]
```diff
@@ -44,14 +40,13 @@ export function useWidgetBorderStyles(
...
-    widgetsEffectedBySpaceDistribution.zones.includes(widgetId);
-  // If the widget is a zone and is distributing space and has no elevated background
-  const isZoneNotElevated = isZoneDistributingSpace && !elevatedBackground;
```
Comment (jsartisan) [inline app/client/src/layoutSystems/anvil/common/hooks/useWidgetBorderStyles.ts:50]: Does it matter if the widget has an elevated background or not to show the border? I removed it thinking it is not relevant check. @riodeuno

**Comment by jsartisan** [comment]
@riodeuno Hey abhinav, before i ask for code review, can you check the gif and tell me if I implemented the correct behaviour.

**Comment by riodeuno** [comment]
@jsartisan We have the design doc for this. 
https://www.figma.com/design/k4mtF0nZGOAanbbb7v1FYN/On-canvas-UI-2.0-(RVMF)-draft?node-id=26-2252&t=BDHPsrjcj60p0Wg1-0

Let me know if this doesn't help. I'll review the state.

**Comment by jsartisan** [comment]
@riodeuno Designs didn't help much. Can you check the DP or the gif.

**Comment by riodeuno** [comment]
@jsartisan 
While redistributing space, the widget name doesn't show up. We can verify this in the `shouldSelectOrFocus` logic in the `anvil/editor/AnvilWidgetName/selectors.ts`.  
``` highlightShown && highlightShown.canvasId === widgetId ``` -> Show widget name if `highlightShown.canvasId` matches the section `widgetId`.
The `AnvilWidgetName/index.tsx` component should render the widget name.

Could you check if this is happening?

File: app/client/src/layoutSystems/anvil/editor/AnvilWidgetName/selectors.ts [hunk 3]
```diff
@@ -61,16 +58,15 @@ export function shouldSelectOrFocus(widgetId: string) {
...
-    getAnvilSpaceDistributionStatus,
     (
       isEditorOpen,
       isDragging,
       highlightShown,
       isWidgetSelected,
       isWidgetFocused,
-      isDistributingSpace,
     ) => {
-      const baseCondition = isEditorOpen && !isDragging && !isDistributingSpace;
```
Comment (jsartisan) [inline app/client/src/layoutSystems/anvil/editor/AnvilWidgetName/selectors.ts:73]: @riodeuno Removed the check of isDistributingSpace here. So it should show the section name when we drag.

**Comment by riodeuno** [comment]
@jsartisan Small note. This change makes the widget name component state to become "select" instead of "focus" so, we see the section widget name in selected state when we are redistributing space. It should be set to "focus" when redistributing. The figma designs show that.

**Comment by jsartisan** [comment]
@riodeuno  
https://github.com/user-attachments/assets/4e90dfd0-8f8c-4d91-82bd-74b1d4f9b2b0

**Comment by riodeuno** [comment]
@jsartisan Yes after we're done redistributing space, the Section should be selected. The screenshare in your previous comment is correct.

**Review by riodeuno — APPROVED** [review]

**Comment by sagar-qa007** [comment]
@jsartisan Could we also verify if there are any test cases that require addition or updates ?

**Pull request merged**